documentation/modules/con-kafka-logging.adoc

```diff
@@ -0,0 +1,89 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-deployment-configuration-kafka.adoc
+
+[id='con-kafka-logging-{context}']
+= Kafka loggers
+
+Kafka has its own configurable loggers:
+
+* `kafka.root.logger.level`
+* `log4j.logger.org.I0Itec.zkclient.ZkClient`
+* `log4j.logger.org.apache.zookeeper`
+* `log4j.logger.kafka`
+* `log4j.logger.org.apache.kafka`
+* `log4j.logger.kafka.request.logger`
+* `log4j.logger.kafka.network.Processor`
+* `log4j.logger.kafka.server.KafkaApis`
+* `log4j.logger.kafka.network.RequestChannel$`
+* `log4j.logger.kafka.controller`
+* `log4j.logger.kafka.log.LogCleaner`
+* `log4j.logger.state.change.logger`
+* `log4j.logger.kafka.authorizer.logger`
+
+Kafka and ZooKeeper use the Apache `log4j` logger implementation.
+
+Use the `logging` property to configure loggers and logger levels.
+
+You can set the log levels by specifying the logger and level directly (inline) or use a custom (external) ConfigMap.
+If a ConfigMap is used, it must be placed under under the `log4j.properties` key according.
+
+Here we see examples of `inline` and `external` logging.
+
+.Inline logging
+[source,yaml,subs="+quotes,attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  # ...
+  logging:
+    type: inline
+    loggers:
+      kafka.root.logger: "INFO"
+  # ...
+  zookeeper:
+    # ...
+    logging:
+      type: inline
+      loggers:
+        zookeeper.root.logger: "INFO"
+  # ...
+  entityOperator:
+    # ...
+    topicOperator:
+      # ...
+      logging:
+        type: inline
+        loggers:
+          rootLogger.level: INFO
+    # ...
+    userOperator:
+      # ...
+      logging:
+        type: inline
+        loggers:
+          rootLogger.level: INFO
+    # ...
+----
+
+.External logging
+[source,yaml,subs="+quotes,attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  # ...
+  logging:
+    type: external
+    name: customConfigMap
+  # ...
+----
+
+Operators use the Apache `log4j2` logger implementation, so the ConfigMap must be placed under `log4j2.properties`.
```

Comment:
One more time even here, it's not clear where is `log4j2.properties` key. It's inside the ConfigMap or it's inside the Kafka resource in the `logging` section. It's not the latter because it's also missing in the example and because ... it's just not :-)

Reply:
Yep. I've rewritten so we say a similar thing to the previous fix.

Operators use the Apache `log4j2` logger implementation, so the logging configuration is described inside the ConfigMap using `log4j2.properties`.